Gitweb:     http://git.kernel.org/linus/96050dfb25966612008dcea7d342e91fa01e993c
Commit:     96050dfb25966612008dcea7d342e91fa01e993c
Parent:     a61d90d75d0f9e86432c45b496b4b0fbf0fd03dc
Author:     Peter Botha <peterb@goldcircle.co.za>
AuthorDate: Tue Jun 9 17:16:32 2009 -0700
Committer:  Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Tue Jun 9 18:09:03 2009 -0700

    char: mxser, fix ISA board lookup
    
    There's a bug in the mxser kernel module that still appears in the
    2.6.29.4 kernel.
    
    mxser_get_ISA_conf takes a ioaddress as its first argument, by passing the
    not of the ioaddr, you're effectively passing 0 which means it won't be
    able to talk to an ISA card.  I have tested this, and removing the !
    fixes the problem.
    
    Cc: "Peter Botha" <peterb@goldcircle.co.za>
    Signed-off-by: Jiri Slaby <jirislaby@gmail.com>
    Acked-by: Alan Cox <alan@lxorguk.ukuu.org.uk>
    Cc: <stable@kernel.org>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 drivers/char/mxser.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/char/mxser.c b/drivers/char/mxser.c
index a420e8d..13f8871 100644
--- a/drivers/char/mxser.c
+++ b/drivers/char/mxser.c
@@ -2711,7 +2711,7 @@ static int __init mxser_module_init(void)
 			continue;
 
 		brd = &mxser_boards[m];
-		retval = mxser_get_ISA_conf(!ioaddr[b], brd);
+		retval = mxser_get_ISA_conf(ioaddr[b], brd);
 		if (retval <= 0) {
 			brd->info = NULL;
 			continue;
